File: generated_code/python/solution_2532_3.py
from typing import List
from collections import Counter

class Solution:
    def equalFrequency(self, word: str) -> bool:
        # Since the length of the word is small (typically <= 100 for this problem),
        # we can simulate the removal of each character one by one.
        # This brute-force approach is highly optimal, straightforward, and avoids complex edge cases.
        for i in range(len(word)):
            # Create a new string with the i-th character removed
            new_word = word[:i] + word[i+1:]
            
            # Count frequencies of the remaining characters
            counts = Counter(new_word)
            
            # If all remaining characters have the exact same frequency,
            # then the set of their unique frequencies will have a size of 1.
            if len(set(counts.values())) == 1:
                return True
                
        return False

File: generated_code/cpp/solution_2532_3.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool equalFrequency(string word) {
        // Step 1: Count frequencies of each character
        vector<int> char_counts(26, 0);
        for (char c : word) {
            char_counts[c - 'a']++;
        }

        // Step 2: Try removing one instance of each possible character
        for (int i = 0; i < 26; ++i) {
            if (char_counts[i] > 0) {
                // Simulate removal
                char_counts[i]--;

                // Check if all remaining non-zero frequencies are equal
                int target_freq = -1;
                bool is_equal = true;
                
                for (int count : char_counts) {
                    if (count > 0) {
                        if (target_freq == -1) {
                            target_freq = count; // Set the baseline frequency
                        } else if (count != target_freq) {
                            is_equal = false;    // Found a mismatch
                            break;
                        }
                    }
                }

                // If valid, return true immediately
                if (is_equal) {
                    return true;
                }

                // Backtrack for the next iteration
                char_counts[i]++;
            }
        }

        return false;
    }
};